File: hardest-language-loop/data/candidates/cand-344a10c7f4/interpreter.py
# interpreter.py for PL-025 L1
# This code defines the programming language used for evaluation.

class CandidateLanguage:
    name = 'PL-025 L1'
    level = 'L1'

# Token-conflict language: parser rewrites conflicting keywords before evaluation
KEYWORD_MAP = {'fn': 'def', 'unless': 'if', 'yieldback': 'return'}


def run_program(source: str, tests: list[dict]) -> dict:
    # TODO: replace with real parser + executor.
    # Current MVP keeps the contract fixed for later integration.
    return {
        'ok': True,
        'candidate': 'cand-344a10c7f4',
        'note': 'MVP interpreter stub; connect real semantics here.'
    }

File: hardest-language-loop/data/candidates/cand-344a10c7f4/interpreter.ml
(* interpreter.ml for PL-025 L1 *)
(* This executable interpreter is the canonical source of truth for the language. *)

exception UndefinedSemantics

(* token-conflict language: conflicting keywords are normalized before evaluation *)
let normalize_keyword k = match k with | "fn" -> "def" | "unless" -> "if" | _ -> k

(* TODO: replace with the real HW3/B-language-derived interpreter body. *)
(* Expected execution path: program.json -> AST -> eval -> output trace *)
